fix: ordering of hpa targets

cyberhck opened the pull request:
When adding a HPA, with memory and cpu limits, then argoCD application appears OutOfSync.

This is because of not consequent sorting the live manifest and the desired manifest.

Fixes #4435

<!-- start metadata -->
---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [x] Changes are compatible[^1]
- [x] Documentation[^2] completed (not needed)
- [ ] Performance impact assessed and acceptable (not sure what I need to do here)
- Tests added and passing[^3] (not sure if this is required)
    - [ ] Unit Tests
    - [ ] Integration Tests
    - [ ] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

apollo-cla commented:
@cyberhck: Thank you for submitting a pull request! Before we can merge it, you'll need to sign the Apollo Contributor License Agreement here: https://contribute.apollographql.com/

cyberhck commented:
please help me get this PR to an acceptable state.

garypen commented:
Interesting bug. The k8s issue that you reference implies this is not a problem with HPA v2. Our chart tries to use v2 if available. Could you please let us know what version of k8s you are encountering this problem with?

garypen commented:
OK, the PR looks good. Please add a changelog as described in (.changesets.README.md) so that your contribution is recognised in our release CHANGELOG.md. Once we have that I'll approve the PR to merge.

cyberhck commented:
@garypen done with a lot of effort :sweat_smile: I hope it's clear and concise, let me know if you'd like any changes. I've also squashed and rebased to provide a cleaner history.

garypen commented:
> @garypen done with a lot of effort 😅 I hope it's clear and concise, let me know if you'd like any changes. I've also squashed and rebased to provide a cleaner history.

Thank you. It's great. Would you mind describing the "lot of effort"? We'd like to make it easy for people to contribute, so any suggestions for improvements that we can make would be welcome.
